src/validate-and-json.js

```diff
@@ -34,47 +33,48 @@ const reportBrokenEdge = (
                 sourceFile,
                 options.rootMarker,
             )}`,
-        });
-        return;
+        };
     }
 
     const NO_CHECKSUM = "No checksum";
     const sourceFileRef = rootRelativePath(
         `${sourceFile}:${sourceLine}`,
         options.rootMarker,
     );
-    jsonItems.push({
+
+    return {
         type: "violation",
         sourceFile: rootRelativePath(sourceFile, options.rootMarker),
-        sourceLine: sourceLine,
+        sourceLine: parseInt(sourceLine, 10),
         targetFile: rootRelativePath(targetFile, options.rootMarker),
-        targetLine: targetLine,
+        targetLine: parseInt(targetLine, 10),
         message: `${sourceFileRef} Updating checksum for sync-tag '${markerID}' referencing '${rootRelativePath(
             targetFile,
             options.rootMarker,
         )}:${targetLine}' from ${
             sourceChecksum || NO_CHECKSUM
         } to ${targetChecksum}.`,
         fix: mappedFix.fix,
-    });
+    };
 };
 
 const validateAndJson = (
     options: Options,
     file: string,
     cache: $ReadOnly<MarkerCache>,
-    jsonItems: Array<JsonItem>,
-): void => {
+): Array<JsonItem> => {
```

Comment:
Longer term, this may be the right thing to do (when a bigger refactor is done), but I think short term, to keep the validate functions inline with one another, the call signatures should match so I think currying the `jsonItems` arg would be the right move in the `getValidator` call.